src/common.c

```diff
@@ -1222,8 +1222,30 @@ int stlink_chip_id(stlink_t *sl, uint32_t *chip_id) {
     int ret;
 
     if (sl->core_id == STM32H7_CORE_ID) {
```

Comment:
This condition check can be removed

Reply:
Yeah, I think we should go through this section and remove the redundant lines and loops.